### ISSUE TITLE
Type representing money values

### ISSUE THREAD

**Issue by rartych** [camaraproject/CarrierBillingCheckOut#105]
**Problem description**
Currently amounts of money are defined as:
```
        amount:
          type: number
          format: float
          multipleOf: 0.01
```
[Swafgger docs](https://swagger.io/docs/specification/data-models/data-types/) indicates 

> `multipleOf` may be used with floating-point numbers, but in practice this can be unreliable due to the limited precision or floating point math.

**Possible evolution**

Storing amount values as long integers is a common best practice

| **Field Name** | **OpenAPI type** | **OpenAPI format** | **Description**                             | 
|----------------|------------------|--------------------|---------------------------------------------|
| `amount`       | `integer`        | `int64`            | Represented in "minor units"                | 
| `currency`     | `string`         | `currency`         | "Minor units" depend on the currency value. | 

**Note**: Minor units are specified according to [ISO 4217](https://en.wikipedia.org/wiki/ISO_4217#Minor_units_of_currency) and can be found in [this table](https://en.wikipedia.org/wiki/ISO_4217#Active_codes).

**Additional context**

Recommendation in:
https://github.com/team-monite/api-style-guide/blob/main/Guidelines.md#should-use-the-common-money-object

> Storing amount values as long integers is a common best practice, adopted by API payment providers like [Adyen](https://docs.adyen.com/development-resources/currency-codes) and [Stripe](https://stripe.com/docs/currencies).

Most currencies have two decimals. Some currencies do not have decimals, and some have three decimals. For example:

-  **GBP 10**: GBP has two decimals, so in minor units submit an amount of 1000
-  **JPY 10**: JPY has no decimals, so in minor units submit an amount of 10
-  **BHD 10**: BHD has three decimals, so in minor units submit an amount of 10000

**Comment by PedroDiez** [comment]
25/10: Moving forward with 
```
        amount:
          type: number
          format: float
          multipleOf: 0.001
```

**Comment by PedroDiez** [comment]
Closed as per [PR#19](https://github.com/camaraproject/CarrierBillingCheckOut/pull/119) MERGED
